src/settings_gui.cpp

```diff
@@ -1066,9 +1067,13 @@ struct GameOptionsWindow : Window {
 			case WID_GO_GUI_SCALE_AUTO:
 			{
 				if (_gui_scale_cfg == -1) {
-					_gui_scale_cfg = _gui_scale;
+					_gui_scale_cfg = previous_gui_scale; // Store the previous GUI scale value
 					this->SetWidgetLoweredState(WID_GO_GUI_SCALE_AUTO, false);
+					if (AdjustGUIZoom(false))
+					    ReInitAllWindows(true);
+					this->gui_scale = _gui_scale;
 				} else {
+					previous_gui_scale = _gui_scale;	// Set the previous GUI scale value as the current one
```

Comment:
```suggestion
					this->previous_gui_scale = _gui_scale; // Store the current GUI scale for later
```

Missing `this->`, and we use spaces, not tabs.